packages/openshift/context.py

```diff
@@ -493,17 +498,24 @@ def tracking(action_handler=None):
     performed, this method will be called with the Action object.
     If not specified, all Actions will aggregate into a internally
     managed Result object which can be accessed with get_result.
+    :param limit: If specified, it allows to specify a limit on the
+    number of actions stored by a given tracking context. If not specified
+    will store unlimited number of oc interactions.
     :return: The tracker contextmanager. If action_handler is not
     specified, call get_result to receive a Result object with all
     tracked Action objects.
     """
     c = Context()
+    if limit:
+        c.limit_tracking_actions = limit
```

Comment:
Adding limit as a constructor argument of Result and passing it in for Result to enforce internally will probably simplify things here. It should also improve encapsulation (i.e. no need to expose `pop_first_action` to external callers).